tests/models/test_model_settings.py

```diff
@@ -0,0 +1,192 @@
+"""Tests for per-model settings functionality."""
+
+from __future__ import annotations
+
+from pydantic_ai import Agent
+from pydantic_ai.messages import ModelMessage, ModelResponse, TextPart
+from pydantic_ai.models.function import AgentInfo, FunctionModel
+from pydantic_ai.models.test import TestModel
+from pydantic_ai.settings import ModelSettings
+
+try:
+    from pydantic_ai.models.gemini import GeminiModel
+
+    gemini_available = True
+except ImportError:  # pragma: no cover
+    GeminiModel = None
+    gemini_available = False
+
+try:
+    from pydantic_ai.models.openai import OpenAIResponsesModel
+
+    openai_available = True
+except ImportError:
+    OpenAIResponsesModel = None
+    openai_available = False
+
+
+def test_model_settings_initialization():
+    """Test that models can be initialized with settings."""
+    settings = ModelSettings(max_tokens=100, temperature=0.5)
+
+    # Test TestModel
+    test_model = TestModel(settings=settings)
+    assert test_model.settings == settings
+
+    # Test FunctionModel
+    def simple_response(messages: list[ModelMessage], agent_info: AgentInfo) -> ModelResponse:
+        return ModelResponse(parts=[TextPart('response')])
+
+    function_model = FunctionModel(simple_response, settings=settings)
+    assert function_model.settings == settings
+
+    agent_info = AgentInfo(function_tools=[], allow_text_output=True, output_tools=[], model_settings=None)
```

Comment:
I think we can skip these 4 lines as they're not testing anything settings related.